### ISSUE TITLE
HelpText.AutoBuild ignores word after last dot in AssemblyTitleAttribute

### ISSUE THREAD

**Issue by ericnewton76** [commandlineparser/commandline#34]
<a href="https://github.com/eakoning"><img src="https://avatars0.githubusercontent.com/u/1950705?v=4" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [eakoning](https://github.com/eakoning)**
_Wednesday Jul 15, 2015 at 14:24 GMT_
_Originally opened as https://github.com/gsscoder/commandline/issues/199_

----

I'm using CommandLine version 1.9.71.2
If i specify an `AssemblyTitleAttribute` with a dotted name (and I version my tool 1.10.1.\* through the AssemblyVersionAttribute) in the project's AssemblyInfo.cs file, e.g.

``` csharp
[assembly: AssemblyTitle("my.favorite.tool")]
```

the Copyright and Heading properties on HelpText when using the AutoBuild show
`my.favorite 1.10.1.27342`

Changing the assembly title to 

``` csharp
[assembly: AssemblyTitle("my.great.fantastic.favorite.tool")]
```

changes Copyright and Heading properties to
`my.great.fantastic.favorite 1.10.1.27342`

It always leaves out the word after the last dot.

When I leave out the `AssemblyTitleAttribute` all together, the HelpText falls back to the text specified in the `AssemblyProductAttribute` and that seems to be working fine.

**Comment by ericnewton76** [comment]
<a href="https://github.com/gsscoder"><img src="https://avatars3.githubusercontent.com/u/1194228?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [gsscoder](https://github.com/gsscoder)**
_Thursday Jul 16, 2015 at 03:25 GMT_

----

@eakoning, can you test it with **2.0**. Changes to public API are easy even if you use it with multiple verb classes.

The effort to maintain two branch is great for us, but if the problem persists in **2.0** I'll be glad to fix it.

Thank you.

**Comment by ericnewton76** [comment]
<a href="https://github.com/scadorel"><img src="https://avatars3.githubusercontent.com/u/2738119?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [scadorel](https://github.com/scadorel)**
_Friday Oct 16, 2015 at 20:50 GMT_

----

Reproduced using latest beta release (2.0.275.0-beta).
Word after the last dot is removed.

The code uses Path.GetFileNameWithoutExtension(titleAttribute.Title) [HeadingInfo.cs] which remove the last word. By default (when creating project in Visual Studio), the project extensions is not given in the assembly info. However, when checking several .net framework assembly and commandline dll, the .dll extension is specified in the assembly title.

```
public static HeadingInfo Default
        {
            get
            {
                var title = ReflectionHelper.GetAttribute<AssemblyTitleAttribute>()
                    .MapValueOrDefault(
                        titleAttribute => Path.GetFileNameWithoutExtension(titleAttribute.Title),
                        ReflectionHelper.GetAssemblyName());
```

```
[assembly: AssemblyTitle("CommandLine.dll")]
```

**Comment by ericnewton76** [comment]
<a href="https://github.com/gsscoder"><img src="https://avatars3.githubusercontent.com/u/1194228?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [gsscoder](https://github.com/gsscoder)**
_Monday Nov 02, 2015 at 06:20 GMT_

----

Hi, @scadorel, as stated I've recently restored (my) support to the project.

I _ping_ you to tell I'll take care of this, for now.

Giacomo

**Comment by ericnewton76** [comment]
<a href="https://github.com/scadorel"><img src="https://avatars3.githubusercontent.com/u/2738119?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [scadorel](https://github.com/scadorel)**
_Thursday Nov 05, 2015 at 23:00 GMT_

----

:+1:

**Comment by ericnewton76** [comment]
<a href="https://github.com/nathan-alden"><img src="https://avatars2.githubusercontent.com/u/1189759?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [nathan-alden](https://github.com/nathan-alden)**
_Sunday Sep 24, 2017 at 15:29 GMT_

----

Still running into this with .NET Core 2.0. `AssemblyTitle` now seems to be automatically generated by Microsoft's project templates and I don't want to turn off auto-generation just to hack `AssemblyTitle` to get it to work with this library.

Optimally, the help text classes should be modified to better support .NET Core 2.0 defaults.

**Comment by ericnewton76** [comment]
<a href="https://github.com/nathan-alden"><img src="https://avatars2.githubusercontent.com/u/1189759?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [nathan-alden](https://github.com/nathan-alden)**
_Sunday Sep 24, 2017 at 15:30 GMT_

----

I modified the `.csproj` and added this element: `<AssemblyTitle>My.App.dll</AssemblyTitle>`, which causes the correct output to appear when CommandLineParser generates help text. I should not have to do this, though.

**Comment by moh-hassan** [comment]
It's resolved in v2.4.3
